modules/ocf_kubernetes/templates/master/loadbalancer/haproxy.cfg.erb

```diff
@@ -20,34 +20,35 @@ defaults
 
 frontend lb-http
   bind 0.0.0.0:80
+  bind :::80
   mode http
 
   # redirect * to *.ocf.berkeley.edu
   http-request redirect prefix https://%[hdr(host)].ocf.berkeley.edu unless { hdr_end(host) -i ocf.berkeley.edu ocf.io }
 
   # https redirect
-  redirect scheme https code 301 if !{ ssl_fc }
+  http-request redirect scheme https code 301 if !{ ssl_fc }
 
 frontend lb-https
   bind 0.0.0.0:443 ssl crt /etc/ssl/private/lb.ocf.berkeley.edu.pem
+  bind :::443 ssl crt /etc/ssl/private/lb.ocf.berkeley.edu.pem
   mode http
 
   # redirect *.ocf.io to *.ocf.berkeley.edu
   http-request redirect prefix https://%[hdr(host),regsub(ocf.io$,ocf.berkeley.edu)] if { hdr_end(host) -i ocf.io }
```

Comment:
we should put `code 301` back. and just hope no one ever sends a host header of xyz.ocf.io:80